### PR TITLE
Result doc year refactor

### PR DESCRIPTION
**Pull request by Stephen-ONeil**
Main point is to get rid of the, ironically, hardcoded dependence on the year templates. ESPECIALLY in the huge amount of results yaml, that stuff is brittle as heck and a real maintenance burden. Probably some other stuff along the way, ~will start a TODO list as I go~.

Absolutely needs to be done before we expand the doc window with the next DPs, but will also make the upcoming DRRs way more straightforward.

TODO:
  - [x] translate `reports_links_text`
  - [x] get more eyes on this, everything seems to be working but can never be too careful with widespread changes

### CODE REVIEW

**Comment by Stephen-ONeil** [comment]
I think I've got the hardcoded use of year templates related to specific results docs cleaned out of all code, just a mountain of yaml left to sort out now.

**Comment by Stephen-ONeil** [comment]
One small translation TODO for the `reports_links_text` key.

**Comment by cookkate** [comment]
Since this was a change that touched a bunch of things would it make sense to recruit another couple of eyes on the page to make sure nothing's broken before we merge it?

File: client/src/panels/panel_declarations/results/drr_summary_text.yaml [hunk 1]
```diff
@@ -1,46 +1,46 @@
...
+      In {{year}}, **{{num_depts}}** organizations sought to achieve **{{fmt_big_int drr_results}}** results.
     {{else}}
-      In {{pa_last_year_2}}, {{subj_name subject}} sought to achieve **{{fmt_big_int drr_results}}** results.
+      In {{year}}, {{subj_name subject}} sought to achieve **{{fmt_big_int drr_results}}** results.
     {{/if}}
     Progress towards meeting these results was measured using **{{fmt_big_int drr_total}}** {{gl_tt "indicators" "IND_RESULT"}}, {{#if drr_past_total}} **{{fmt_big_int drr_past_total}}**
-    of which were seeking to achieve their target **by** March 31, {{pa_last_year_2_short_second}}{{#if drr_future_total}} and{{else}}.{{/if}}
-    {{/if}}{{#if drr_future_total}} **{{fmt_big_int drr_future_total}}** of which are seeking to achieve their target **after** March 31, {{pa_last_year_2_short_second}}
+    of which were seeking to achieve their target **by** March 31, {{target_year}}{{#if drr_future_total}} and{{else}}.{{/if}}
+    {{/if}}{{#if drr_future_total}} **{{fmt_big_int drr_future_total}}** of which are seeking to achieve their target **after** March 31, {{target_year}}
```
[the remark this answers is not in the document]
Comment (Stephen-ONeil) [reply client/src/panels/panel_declarations/results/drr_summary_text.yaml]: Could you re-review that text now that I've tweaked the bolding around? Thanks!

[the remark this answers is not in the document]
Comment (cookkate) [reply client/src/panels/panel_declarations/results/drr_summary_text.yaml]: Looks good, thanks!

**Pull request merged**